operator/v1/types_console.go

```diff
@@ -99,8 +99,59 @@ type StatuspageProvider struct {
 	PageID string `json:"pageID"`
 }
 
+// WidgetName defines names of UI widgets in the console UI.
+type WidgetName string
+
+const (
+	// lightspeedButton is the name for the Lightspeed button HTML element.
+	LightspeedButton WidgetName = "LightspeedButton"
+)
+
+// WidgetState defines the state of the widget in the console UI.
+type WidgetState string
+
+const (
+	// "Show" means that the widget will be rendered in the console UI.
+	WidgetShow WidgetState = "Show"
+	// "Hide" means that the widget will not be rendered in the console UI.
+	WidgetHide WidgetState = "Hide"
+)
+
+// WidgetVisibility defines the criteria to show/hide a widget.
+// +union
+type WidgetVisibility struct {
+	// state defines the if the widget is shown or hidden in the console UI.
+	// Showing the widget in the console UI is represented by the "Show" value.
+	// Hidding the widget in the console UI is represented by the "Hide" value.
+	// +unionDiscriminator
+	// +kubebuilder:validation:Enum:="Show";"Hide"
+	// +kubebuilder:validation:Required
+	State WidgetState `json:"state"`
+}
+
+// Widgets contains set of UI widgets and their state in the console UI.
+type Widget struct {
+	// name is the unique name of an widget.
+	// Available widgets are LightspeedButton.
+	// +kubebuilder:validation:Enum:="LightspeedButton";
+	// +kubebuilder:validation:Required
+	Name WidgetName `json:"name"`
+	// visibility defines the visibility state of widget.
+	// +kubebuilder:validation:Required
+	Visibility WidgetVisibility `json:"visibility"`
+}
+
 // ConsoleCustomization defines a list of optional configuration for the console UI.
 type ConsoleCustomization struct {
+	// widgets defines an array of widgets that can be interacted with in the console UI.
```

Comment:
Is there anything that we can add for a user to understand what we mean by a widget? A brief extra line in the description to show what kind of thing we mean?

Reply:
I wonder if we should make something even more generic like "capability"

Reply:
the thing is that narrowing down the state just to `enabled/disabled` or even `hide/show` is too bold, since "widget" itself could be any HTML element or even a React component, so there could be more to it then just the two mentioned...

Reply:
By "capability," I was just suggesting a different name for the property. Maybe that is confusing since "capability" is also used for enabling/disabling capabilities during install.

Reply:
Capability could work, how would you define the lightspeed button as a capability? I think we want something that will clearly describe what the things within it are, I just am not sure how you'd define a widget right now (I think we had similar conversations for perspectives?)

Reply:
@spadgett what about `Component` or `Element` ?

Reply:
My thinking is that we probably won't have many specific elements/components on the page we want to allow users to show and hide. If we make the name a little more generic, we could use it for enabling or disabling broader console features. The Lightspeed button itself isn't a capability, but Lightspeed support in console is. So it could be something like

```yaml
capabilities:
  enabled:
  - QuickStarts
  - Plugins
  disabled:
  - Lightspeed
```

Not sure if this is too similar to feature gates, though.

Reply:
@spadgett With that thought, which seems reasonable to me, do you think a binary state of enabled/disabled for each of these would work, or do you think having something structured where we could add some conditions if we needed to (like perspectives has) would be advantageous?

Reply:
@JoelSpeed Being able to extend it later with something structured is a good idea 👍 How do you think that would look?

Reply:
For now we just make it effectively the same as the perspectives
```
capabilities:
- name: Foo | Bar | Baz
  visibility:
    state: Show | Hide | BasedOnMyCondition
    basedOnMyCondition: {}
```

Reply:
@spadgett ok I see that you mean now with the "Capabilities"...  what about `enhancements`? 

> .... do you think having something structured where we could add some conditions if we needed to (like perspectives has) would be advantageous?

@JoelSpeed  I dont think we need to have something structured, we could even downscae lthe PR to what @spadgett [proposed](https://github.com/openshift/api/pull/1956#discussion_r1681165702), with having a space to extend the capabilities/enhancements fields (`enabled/disabled`).

Reply:
How would you extend the Hide or Show semantic of this without having a structured object list?

Reply:
@JoelSpeed sorry for the confusion I've a bit misunderstood what you meant in your [comment](https://github.com/openshift/api/pull/1956#discussion_r1683886290).
If we all align on the structure (looks god to me 👍 ), do we wanna:keep the name of t `capabilities`

Reply:
@JoelSpeed sorry for the confusion I've a bit misunderstood what you meant in your [comment](https://github.com/openshift/api/pull/1956#discussion_r1683886290).
If we all align on the structure (looks god to me 👍 ), do we wanna:
1. keep the name of top field `capabilities`? I like the proposed `widgets`, but if we want to make it feature based, the I like `enhancements`. WDYT ?
2. keep the states as they are (`Show/Hide`) or reuse what perspective has (`enabled/disabled`) ?

Reply:
I would probably align to features (calling capabilities), and use enabled/disabled.

Are we concerned about the potential conflict with the generic capabilities in the install configuration? 🤔

Reply:
> Are we concerned about the potential conflict with the generic capabilities in the install configuration? 🤔

Reply:
> Are we concerned about the potential conflict with the generic capabilities in the install configuration? 🤔

I would assume no, since they are set in different configs (CVO <-> ConsoleOperator). I could see that some capability names could be identical, but hard to tell. We could prefix it with something like `uiCapabilities` or `frontendCapabilities` to rule out potential conflicts?